### PR TITLE
(Proposal) Added a note on forms validation

### PR DESCRIPTION
**Pull request by mickaelandrieu**
Hello!

regarding the way I validate my forms, I've always found that storing validation rules into my entities is a better idea than using the constraints inside my form types.

When I need a really specific control on my form validation, I inject validation groups: do you think it can be viewed as a best practice?

### CODE REVIEW

File: best_practices/forms.rst [hunk 1]
```diff
@@ -137,6 +137,27 @@ view layer:
...
+
+    class Post
+    {
+       /**
+         * @Assert\NotBlank()
+         */
+         public $title;
+     }
+
+This make them reusables and independents from the validation rules.
```
Comment (xabbuh) [inline best_practices/forms.rst]: independent

File: best_practices/forms.rst [hunk 2]
```diff
@@ -137,6 +137,27 @@ view layer:
         <input type="submit" class="btn" value="Create" />
     {{ form_end(form) }}
 
+Validation
+----------
+
+Define your validation constraints outside of the forms, for instance directly on the object
+class used as data mapper::
```
Comment (xabbuh) [inline best_practices/forms.rst]: Maybe reword it a bit like this:

> Do not define your validation constraint with the `constraints` option. Configure them on the object the form is mapped to instead:

**Comment by xabbuh** [comment]
👍 I agree

**Comment by javiereguiluz** [comment]
@mickaelandrieu I liked both your original proposal and the reword suggestions made by @xabbuh so I reworded this accordingly. What do you think? Thanks!

**Comment by mickaelandrieu** [comment]
@javiereguiluz you have improved my proposal as always!

Thank you :+1:

File: best_practices/forms.rst [hunk 3]
```diff
@@ -137,6 +137,34 @@ view layer:
         <input type="submit" class="btn" value="Create" />
     {{ form_end(form) }}
 
+Validation
+----------
+
+The `constraints`_ option allows you to attach `validation constraints`_ to any
```
Comment (xabbuh) [inline best_practices/forms.rst]: we should use the `doc` or `ref` role here instead
